packages/databyss-editor/state/EditorProvider.tsx

```diff
@@ -132,10 +137,41 @@ const EditorProvider: React.FunctionComponent<PropsType> = ({
       payload: { id },
     })
 
+  const copy = (e: ClipboardEvent) => {
+    const _frag = getCurrentSelection(state)
+
+    // TODO: set plain text
+    e.clipboardData.setData('text/plain', 'Hello, world!')
+
+    // set application data for clipboard
+    e.clipboardData.setData('application/databyss-frag', JSON.stringify(_frag))
+
+    // TODO: SET HTML
+    dispatch({
+      type: COPY,
+      payload: {},
```

Comment:
if `payload` is removed the reducer will throw an error at `let nextSelection = payload.selection` even if the reducer doesn't catch the `COPY` action in the switch statement.

Reply:
Ah ok, we should probably change that to `let nextSelection = payload?.selection` in the reducer to avoid this. According to the [FSA spec](https://github.com/redux-utilities/flux-standard-action#actions), payload should be optional.